LoadRunner Professional Integration Tile

mihneavelinOT opened the pull request:
### What does this PR do?

Create LoadRunner Professional Integration tile.

### Motivation

LoadRunner Professional will have a bi-directional data-feed with Datadog

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Overview
+
+OpenText™ LoadRunner Professional provides a load testing solution that enables you to test the performance of diverse application types, and to identify and resolve issues before applications go live.
```
```suggestion
OpenText™ LoadRunner Professional is a load testing solution that enables you to test the performance of diverse application types, and identify and resolve issues before applications go live.
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+LoadRunner Controller is a tool for creating and controlling LoadRunner Professional scenarios. A scenario defines the events that occur during each testing session. It controls the number of users to emulate (virtual users, or Vusers), the actions they perform, and the machines on which they run their emulations. You use scenarios to create load tests to check the reliability and strength of you ...
+
+You can configure Controller to push real-time metrics and data from scenario runs to Datadog.
```
```suggestion
With this integration, Controller pushes real-time metrics and data from scenario runs to Datadog.
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+LoadRunner Controller is a tool for creating and controlling LoadRunner Professional scenarios. A scenario defines the events that occur during each testing session. It controls the number of users to emulate (virtual users, or Vusers), the actions they perform, and the machines on which they run their emulations. You use scenarios to create load tests to check the reliability and strength of you ...
+
+You can configure Controller to push real-time metrics and data from scenario runs to Datadog.
+
```
Could you add another sentence or two here describing the metrics and data that's being sent in? i.e. is it performance data, is it metrics/logs/alerts/something else?

mihneavelinOT replied:
i added the table from below

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Setup
+
+Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
```
It would be helpful to elaborate in the Overview section on what scenario information vs. run metrics are

mihneavelinOT replied:
added the table from below to Overview section

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Setup
+
+Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
```
```suggestion
Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. Once configured, this integration provides a Datadog dashboard to view the data in pre-configured widgets.
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
+
+1.	In the Controller toolbar, select __Tools > Datadog Configuration__.
```
Since users will be reading this from the Datadog app (or Datadog docs) it would be useful to add another first step here to navigate to the LoadRunner platform (and a link).

mihneavelinOT replied:
LoadRunner Controller is a desktop application, maybe a first step saying something like "open Controller"?

anawishnoff replied:
Yes that would be helpful!

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
+
+1.	In the Controller toolbar, select __Tools > Datadog Configuration__.
+2.	In the __Site__ field, select the region defined in your Datadog account.
```
```suggestion
2.	In the __Site__ field, select your [Datadog site][6].
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+[2]: https://raw.githubusercontent.com/DataDog/integrations-extras/master/loadrunner_professional/images/datadog_configuration_window.png
+[3]: https://docs.datadoghq.com/help/
+[4]: https://admhelp.microfocus.com/lr
+[5]: https://github.com/DataDog/integrations-extras/blob/master/loadrunner_professional/metadata.csv
```
```suggestion
[5]: https://github.com/DataDog/integrations-extras/blob/master/loadrunner_professional/metadata.csv
[6]: https://docs.datadoghq.com/getting_started/site/
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+1.	In the Controller toolbar, select __Tools > Datadog Configuration__.
+2.	In the __Site__ field, select the region defined in your Datadog account.
+3.	In the __API Key__ field, enter the API key generated by Datadog.
```
May want to link to our docs here on how to create an api key: https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+4.	Click __Test Connection__.
+5.	Once the connection is successful, select whether to send scenario information and/or run information to Datadog:
+
+|   |   |
+|---|---|
+|__Send scenario information__| Send information about the scenario run, such as the start and stop time, duration, and included scripts. |
+|__Send run information__| Send metrics from the scenario run, such as Vuser status and transaction response times. |
```
This would be helpful to add in the Overview section instead

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+|__Send scenario information__| Send information about the scenario run, such as the start and stop time, duration, and included scripts. |
+|__Send run information__| Send metrics from the scenario run, such as Vuser status and transaction response times. |
+	
+6.	If you enable Controller to send scenario information, install the __LoadRunner Run Info Pipeline__ in Datadog to ensure that tags are applied to the scenario information.
```
How does a user install the LoadRunner Run Info Pipeline?

mihneavelinOT replied:
I am refering to the LoadRunner log pipeline that will be available in DataDog here. Maybe put the link form Datadog Docs on how to install a log pipeline?

anawishnoff replied:
Ah i see - updated my suggestion below

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+|__Send run information__| Send metrics from the scenario run, such as Vuser status and transaction response times. |
+	
+6.	If you enable Controller to send scenario information, install the __LoadRunner Run Info Pipeline__ in Datadog to ensure that tags are applied to the scenario information.
+7.	Optionally, in Datadog, install the __LoadRunner Stats__ dashboard. The dashboard includes widgets that display run information and scenario information (depending on the data that Controller is configured to send). 
```
```suggestion
7. In Datadog, the __LoadRunner Stats__ dashboard will automatically be installed with the integration. The dashboard includes widgets that display run information and scenario information (depending on the data that Controller is configured to send). 
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+	
+6.	If you enable Controller to send scenario information, install the __LoadRunner Run Info Pipeline__ in Datadog to ensure that tags are applied to the scenario information.
+7.	Optionally, in Datadog, install the __LoadRunner Stats__ dashboard. The dashboard includes widgets that display run information and scenario information (depending on the data that Controller is configured to send). 
+When you run a Controller scenario, the data is displayed in the dashboard widgets.
```
```suggestion
```

Don't think this sentence is needed with context above

anawishnoff commented on loadrunner_professional/assets/dashboards/loadrunner_professional_overview.json:
```diff
@@ -0,0 +1,280 @@
+{
+    "title": "LoadRunner Professional Dashboard",
+    "description": "LoadRunner Professional Dashboard shows the running vusers, transactions statistics and previous Controller runs. For detailed transactions statistics the user can filter after the trasaction's name tag.",
```
```suggestion
    "description": "LoadRunner Professional Dashboard shows the running vusers, transactions statistics and previous Controller runs. For detailed transactions statistics the user can filter by the transaction's name tag.",
```

anawishnoff commented on loadrunner_professional/assets/dashboards/loadrunner_professional_overview.json:
Please see our [dashboard best practices guide](https://docs.datadoghq.com/developers/integrations/create-an-integration-dashboard). Specifically, please add:
- a logo onto the dashboard
- text widgets that provide helpful documentation or further context on what is being shown

mihneavelinOT replied:
i added the logo in a banner with a description under it

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,59 @@
+    "changelog": "CHANGELOG.md",
+    "description": "Send your LoadRunner Professional metrics to DataDog.",
+    "title": "LoadRunner Professional",
+    "media": [
+      {
+        "media_type": "image",
+        "caption": "Datadog Configuration Window",
+        "image_url": "images/datadog_configuration_window.png"
+      }
```
This can be removed unless you want this image to show up in the media carousel on your tile.

Instead, please add a few images of your Datadog dashboard, and/or any relevant screenshots of your platform that you'd like to display on your tile.

mihneavelinOT replied:
i set the "banner_img": "https://raw.githubusercontent.com/DataDog/integrations-extras/master/loadrunner_professional/assets/logos/loadrunner_logo_4x3.png", the logo will not be shown until the integration in released, any workaround?

mihneavelinOT replied:
"Kubernetes Cluster Overview Dashboard" has this in the "Image URL" field of the banner: "/static/images/integration_dashboard/kubernetes_hero_2.jpeg"

anawishnoff replied:
Just checked with the team - if you'd like to host your image on GitHub and display it on your dashboard, the link will be the following once the PR is merged:

https://github.com/DataDog/integrations-extras/tree/master/loadrunner_professional/<path-to-image>

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,59 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Send your LoadRunner Professional metrics to DataDog.",
```
I would suggest to make this slightly more specific and mention the types of data being sent in

anawishnoff reviewed (changes requested):
first round of comments from my end. thanks!

buraizu commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Setup
+
+Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information and/or run metrics. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
```
```suggestion
Configure LoadRunner Controller to push data to Datadog. You can choose whether to send scenario information, run metrics, or both. In Datadog, you can install a dashboard to view the data in pre-configured widgets.
```

buraizu replied:
Adding this suggestion for accessibility. If possible, could you please similarly update anywhere that uses `and/or` to clarify?

buraizu commented on loadrunner_professional/metadata.csv:
```diff
@@ -0,0 +1,10 @@
+loadrunner.vusers.ready,gauge,,vuser,,Number of vusers ready to run.,0,loadrunner_professional,vusers.ready,
+loadrunner.vusers.finished,gauge,,vuser,,Number of vusers that have finished running.,0,loadrunner_professional,vusers.finished,
+loadrunner.vusers.error,gauge,,vuser,,Number of vusers that have failed with error.,0,loadrunner_professional,vusers.error,
+loadrunner.total.transactions.passed.per.sec,gauge,,transaction,second,Total transactions passed per second - Controller wise.,0,loadrunner_professional,total.transactions.passed.per.sec,
```
```suggestion
loadrunner.total.transactions.passed.per.sec,gauge,,transaction,second,Total transactions passed per second for the LoadRunner Controller.,0,loadrunner_professional,total.transactions.passed.per.sec,
```

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,57 @@
+3.	In the __API Key__ field, enter the [API key][7] generated by Datadog.
+4.	Click __Test Connection__.
+5.	Once the connection is successful, select whether to send scenario information, run information, or both to Datadog.
+6.	If you enable Controller to send scenario information, install the __LoadRunner Run Info Pipeline__ in Datadog to ensure that tags are applied to the scenario information.
```
```suggestion
6. If you enable Controller to send scenario information, the included log pipeline with this integration will automatically process your logs and add relevant tags. For more specifics on the pipeline, navigate to Logs > Pipelines in Datadog.
```

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,54 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Send your LoadRunner Professional metrics from the scenario run, such as Vuser status and transaction response times, information about the scenario run, such as the start and stop time, duration, and included scripts, or both.",
```
Sorry - looks like this is over the 120 character limit, please shorten. Thanks!

anawishnoff commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,57 @@
+
+|   |   |
+|---|---|
+|__Send scenario information__| Send information about the scenario run, such as the start and stop time, duration, and included scripts. |
```
Is the "information" described here sent in the form of logs? If so, i'd mention that explicitly. If not, would be helpful to mention in the Overview what the value of the submitted logs are

anawishnoff reviewed (changes requested):
A few more comments from my end, thanks!

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+id: loadrunner-professional
```
The id needs to match the name of the file. In this case the file should be called `loadrunner-professional` (or the other way around, the id of the integration should be `loadrunner_professional`).

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,77 @@
+id: "loadrunner-professional"
```
Same as before the file should be renamed with the `<integrationid>_tests.yaml`.

That should trigger the execution of the tests.

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,77 @@
+tests:
+  - sample: >+
+        {
+      		"host": "controller_host",
```
Is this the format that is sent to datadog's intake ?

It looks like the output that the datadog api returns when logs are queried. See this [doc](https://docs.datadoghq.com/developers/integrations/log_integration/#review-and-deploy-the-integration)

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,54 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Send your LoadRunner Professional metrics from the scenario run, information about the scenario run, or both.",
```
```suggestion
    "description": "Send LoadRunner Professional metrics and information about scenario runs to Datadog",
```

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,54 @@
+    "changelog": "CHANGELOG.md",
+    "description": "Send your LoadRunner Professional metrics from the scenario run, information about the scenario run, or both.",
+    "title": "LoadRunner Professional",
+    "media": [
```
Please add at least one image here - we recommend starting with a screenshot of your dashboard, but you can also add any relevant screenshots of your platform or marketing media as well to give the user an idea of what their experience with the integration will look like.

mihneavelinOT replied:
Like I said earlier, I cannot take a screenshot of the final version of the dashboard since the logo from the banner is not visible.
I added some screenshots of the product instead. And if possible, I will add a screenshot of the dashboard after the integration tile goes live.

anawishnoff replied:
Thanks - that works for now

anawishnoff reviewed (changes requested):
A few more comments from my end but overall this is looking good. Thanks!

bgoldberg122 commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,79 @@
+    "name": "OpenText",
+    "homepage": "https://www.microfocus.com/en-us/products/loadrunner-professional/overview",
+    "sales_email": "dmcleish@opentext.com"
+  },
+  "oauth": {}
+}
```
```suggestion
  }
}
```

anawishnoff commented on loadrunner_professional/manifest.json:
```diff
@@ -0,0 +1,79 @@
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Send LoadRunner Professional metrics and information about scenario runs to Datadog.",
```
```suggestion
    "description": "Send LoadRunner Professional metrics and information about scenario runs to Datadog",
```

anawishnoff reviewed (approved):
One small styling nit otherwise LGTM. Thanks!

drichards-87 commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+3.	In the __Site__ field, select your [Datadog site][6].
+4.	In the __API Key__ field, enter the [API key][7] generated by Datadog.
+5.	Click __Test Connection__.
+6.	Once the connection is successful, select whether to send scenario information, run information, or both to Datadog.
```
```suggestion
6.	Once the connection is successful, choose whether to send scenario information, run information, or both to Datadog.
```

drichards-87 commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+4.	In the __API Key__ field, enter the [API key][7] generated by Datadog.
+5.	Click __Test Connection__.
+6.	Once the connection is successful, select whether to send scenario information, run information, or both to Datadog.
+7.	If you enable Controller to send scenario information, the included log pipeline with this integration will automatically process your logs and add relevant tags. For more specifics on the pipeline, navigate to Logs > Pipelines in Datadog.
```
```suggestion
7.	If you enable Controller to send scenario information, the included log pipeline with this integration automatically processes your logs and adds relevant tags. For more specifics on the pipeline, navigate to Logs > Pipelines in Datadog.
```

drichards-87 commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+5.	Click __Test Connection__.
+6.	Once the connection is successful, select whether to send scenario information, run information, or both to Datadog.
+7.	If you enable Controller to send scenario information, the included log pipeline with this integration will automatically process your logs and add relevant tags. For more specifics on the pipeline, navigate to Logs > Pipelines in Datadog.
+8.	In Datadog, the __LoadRunner Stats__ dashboard will automatically be installed with the integration. The dashboard includes widgets that display run information and scenario information (depending on the data that Controller is configured to send).
```
```suggestion
8.	In Datadog, the __LoadRunner Stats__ dashboard is automatically installed with the integration. The dashboard includes widgets that display run information and scenario information (depending on the data that Controller is configured to send).
```

drichards-87 commented on loadrunner_professional/README.md:
```diff
@@ -0,0 +1,58 @@
+
+## Overview
+
+OpenText™ LoadRunner Professional is a load testing solution that enables you to test the performance of diverse application types, and identify and resolve issues before applications go live.
```
```suggestion
OpenText™ LoadRunner Professional is a load testing solution that enables you to test the performance of diverse application types to identify and resolve issues before applications go live.
```

drichards-87 reviewed (approved):
Left a couple of suggestions and approved the PR.

moreauowen commented:
👋🏻 Hi there! Looks like `source_type_id` is missing from the `assets`->`integration` object in the manifest.

To generate it, please do the following:
- Update `ddev` to the latest version
- Run `ddev create ...` as you previously did, the new manifest created will automatically include a `source_type_id` field
- Copy that field into the manifest.json in this PR

At this point, CI validations should pass and I will take another look. Thank you!

mihneavelinOT commented:
Hi @moreauowen,

I followed the mentioned steps.

Updated ```ddev``` from ```6.2.0```:
```ddev --version``` printed ```ddev, version 8.0.0``` after the upgrade

Deleted the existing integration folder from the repo and then entered the following command:
```ddev create -t tile "LoadRunner Professional"```

This is what I have now in the ```assets``` object:
```
"assets": {
    "dashboards": {},
    "monitors": {},
    "logs": {},
    "saved_views": {}
  }
```

No ```source_type_id``` was added. Any ideas? Thank you!

moreauowen commented:
Hi there @mihneavelinOT,

Sorry for the confusion. When running `ddev create ...` please use a type other than `tile`.
```
ddev create -t check "LoadRunner Professional"
```

This will automatically create the `source_type_id`. You can then copy that field into the manifest in this PR.

Thank you!

mihneavelinOT commented:
It worked, thanks a lot @moreauowen !

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+        - scenario_started.general.version
+        - scenario_ended.general.version
+      sourceType: attribute
+      target: lr_version
```
Can you please remap to the standard datadog `version` tag ? See datadog standard attributes [doc](https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/)

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+      sources:
+        - scenario_started.summary.mode
+      sourceType: attribute
+      target: lr_scenario_mode
```
If the idea is to identify this attribute as specific to load runner. Can we namespace with `loadrunner` instead of `lr` that is both less readable for users and can also describe other technologies.

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+      sources:
+        - scenario_started.summary.load_behavior
+      sourceType: attribute
+      target: lr_scenario_load_behavior
```
Same as before
```suggestion
      target: loadrunner.scenario.load_behavior
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+      name: LoadRunner Hostname Tag Creation
+      enabled: true
+      sources:
+        - hostname
```
It would be preferable to map this to the standard `host` tag.

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,71 @@
+      sources:
+        - service
+      sourceType: attribute
+      target: lr_service
```
Why do we want to drop the `service` attribute and remap it to a custom `lr_service` tag ?

mihneavelinOT replied:
removed the remapper entirely

emarsha94 commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,61 @@
+id: loadrunner_professional
+metric_id: loadrunner_professional
+facets: null
```
Can you confirm you don't want facets out of the box with your integration? For context, a log pipeline will help customers easily identify that the logs are coming from your integration, and will include out of the box facets to help customers understand what attributes are important to the integration and for them to quickly filter their logs.

mihneavelinOT replied:
Hi @emarsha94, we confirm we do not want facets. All the related tags for the logs will be created via the pipeline.

emarsha94 commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,16 @@
+tests:
+  - sample: >+
+        {"scenario_started":{"general":{"product": "LoadRunner Controller","version": "2024.1.0.0","path": "Scenario1","resultname": "","result_file": "C:\\Users\\user1\\AppData\\Local\\Temp\\2\\res\\res.lrr","start_time": 1707817084.000000,"daylight_bias": "0 minutes","time_zone": "0 seconds"},"scripts":[{"name": "webhttphtml1","type": "Multi+QTWeb","path": "C:\\Users\\user1\\Desktop\\WebHttpHtm ...
+      	
```
```suggestion
```

emarsha94 commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,16 @@
+    result: null
+  - sample: |+
+        {"scenario_ended":{"execution":{"start_time": 1707817084.000000,"stop_time": 1707817214.000000,"total_duration": "130 seconds"},"general":{"product": "LoadRunner Controller","version": "2024.1.0.0","path": "Scenario1","resultname": "","result_file": "C:\\Users\\user1\\AppData\\Local\\Temp\\2\\res\\res.lrr","start_time": 1707817084.000000,"daylight_bias": "0 minutes","time_zone": "0 second ...
+
+      	
```
```suggestion
```

emarsha94 commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,16 @@
+
+      	
+    result: null
+
```
```suggestion
```

emarsha94 reviewed (changes requested):
Please remove all empty lines and tabs from the /loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml file. I made suggestions for you to commit to make it easier :)

emarsha94 commented on loadrunner_professional/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - LoadRunner Professional
+
+## 1.0.0 / 2024-01-18
```
```suggestion
## 1.0.0 / 2024-03-02
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,100 @@
+        }
+      tags:
+      - "source:LOGS_SOURCE"
+# The `result` field should be left blank to start. Once you submit your log asset files with
+# your integration pull-request in a Datadog GitHub repository, Datadog's validations will
+# run your raw logs against your pipeline and return the result. If the result output in the
+# validation is accurate, take the output and add it to the `result` field in your test YAML file.
```
```suggestion
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+        - scenario_started.summary.mode
+      sourceType: attribute
+      target: loadrunner.scenario.mode
+      targetType: tag
```
Is there any reason to keep this as a tag ? While host and version are standard tags, I get the impression that the other attributes being remapped here look more like attributes (tags are mostly infra related)
```suggestion
      targetType: attribute
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+        - scenario_started.summary.scenario_type
+      sourceType: attribute
+      target: loadrunner.scenario.type
+      targetType: tag
```
```suggestion
      targetType: attribute
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+        - scenario_started.summary.load_behavior
+      sourceType: attribute
+      target: loadrunner.scenario.load_behavior
+      targetType: tag
```
```suggestion
      targetType: attribute
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,100 @@
+id: "loadrunner_professional"
+tests:
+  - sample: >+
```
nit:

Do you mind using `|-` as a separator. It works the same but `|-` seems more commonly used for other integrations.
```suggestion
  - sample: |-
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional_tests.yaml:
```diff
@@ -0,0 +1,100 @@
+        }
+      tags:
+      - "source:LOGS_SOURCE"
+  - sample: |+
```
```suggestion
  - sample: |-
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+id: loadrunner
+metric_id: loadrunner-professional
+backend_only: false
+facets: null
```
```suggestion
facets: []
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+    query: "source:loadrunner"
+  processors:
+    - type: attribute-remapper
+      name: Map attribute `hostname` to tag `host`
```
I don't see the `hostname` attribute in any of the samples. Is it because it's provided outside of the message field or because we're missing a representative sample ?

mihneavelinOT replied:
Indeed, it is provided outside the message field.

emarsha94 commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+  name: LoadRunner Scenario Information Pipeline
+  enabled: true
+  filter:
+    query: "source:loadrunner"
```
```suggestion
    query: "source:loadrunner_professional"
```

txominpelu commented on loadrunner_professional/assets/logs/loadrunner_professional.yaml:
```diff
@@ -0,0 +1,62 @@
+  filter:
+    query: "source:loadrunner_professional"
+  processors:
+    - type: attribute-remapper
+      name: Map attribute `hostname` to tag `host`
+      enabled: true
+      sources:
+        - hostname
+      sourceType: attribute
+      target: host
+      targetType: tag
+      preserveSource: false
+      overrideOnConflict: false
```
```suggestion
```

emarsha94 commented on loadrunner_professional/CHANGELOG.md:
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - LoadRunner Professional
+
+## 1.0.0 / 2024-03-02
```
```suggestion
## 1.0.0 / 2024-03-14
```

txominpelu commented:
I've updated the branch, merging master into it to bring the `CODEOWNERS` changes where logs backend owns the logs assets.

txominpelu commented on .github/CODEOWNERS:
Do you mind adding `@DataDog/logs-backend` as owners of `/loadrunner_professional/assets/logs`

mihneavelinOT replied:
Not at all
